test/common/network/udp_listener_impl_test.cc

```diff
@@ -144,12 +160,12 @@ TEST_P(UdpListenerImplTest, UseActualDstUdp) {
   EXPECT_CALL(listener_callbacks_, onReadReady());
   EXPECT_CALL(listener_callbacks_, onData(_))
       .WillOnce(Invoke([&](const UdpRecvData& data) -> void {
-        validateRecvCallbackParams(data);
+        validateRecvCallbackParams(data, Api::OsSysCallsSingleton::get().supportsMmsg() ? 16u : 1u);
```

Comment:
Maybe add a validation function that can avoid some of this boilerplate.

Reply:
This was discussed in the intial review with Dan. 
We had decided that we would like to add `num_packet_per_recv` to `validateReceiveCallbackParams` to allow more control over the function with the caller.